internal/convert.go

```diff
@@ -373,38 +419,66 @@ func (conv *Conv) getStatementStat(s string) *statementStat {
 	return conv.stats.statement[s]
 }
 
-func (conv *Conv) skipStatement(l []nodes.Node) {
-	if conv.schemaMode() { // Record statement stats on first pass only.
+func (conv *Conv) SkipStatement(l []nodes.Node) {
```

Comment:
Yeah, the existing code here is kind of broken: we shouldn't reference nodes.Node in convert.go.
All of that should live in pgdump.go.

I think what should happen longer-term is that prNodes and prNodeType should be moved into pgdump.go.

Let's go with what you have for the moment, but could you add a TODO something like:

// TODO: Move prNodes to pgdump.go, remove all references to nodes.Node in convert.go,
// and just have one version of SkipStatement, ErrorInStatement, SchemaStatement and DataStatement.

Reply:
I had created new functions with MySQL suffix because using one version needed changes in pgdump.go.
Sure, now you can check in new commit that I have moved prNodes and prNodeType to pgdump.go and created one version for SkipStatement, ErrorInStatement, SchemaStatement and DataStatement functions.